README.md

```diff
@@ -1,77 +1,150 @@
+  statements and procedures.
 
 ## Role Variables
 
-A description of all input variables (i.e. variables that are defined in
-`defaults/main.yml`) for the role should go here as these form an API of the
-role.
+### `mssql_accept_microsoft_odbc_driver_17_for_sql_server_eula`
 
-Variables that are not intended as input, like variables defined in
-`vars/main.yml`, variables that are read from other roles and/or the global
-scope (ie. hostvars, group vars, etc.) can be also mentioned here but keep in
-mind that as these are probably not part of the role API they may change during
-the lifetime.
+Set this variable to `true` to indicate that you accept EULA for installing the
+`msodbcsql17` package.
 
-Example of setting the variables:
+The license terms for this product can be downloaded
+from <https://aka.ms/odbc17eula> and found in `/usr/share/doc/msodbcsql17/LICENSE.txt`.
 
-```yaml
-template_foo: "oof"
-template_bar: "baz"
-```
+Default: `false`
 
-### Variables Exported by the Role
+### `mssql_accept_microsoft_cli_utilities_for_sql_server_eula`
 
-This section is optional.  Some roles may export variables for playbooks to
-use later.  These are analogous to "return values" in Ansible modules.  For
-example, if a role performs some action that will require a system reboot, but
-the user wants to defer the reboot, the role might set a variable like
-`template_reboot_needed: true` that the playbook can use to reboot at a more
-convenient time.
+Set this variable to `true` to indicate that you accept EULA for installing the
+`mssql-tools` package.
 
-Example:
+The license terms for this product can be downloaded
+from <http://go.microsoft.com/fwlink/?LinkId=746949> and found in
+`/usr/share/doc/mssql-tools/LICENSE.txt`.
 
-`template_reboot_needed` - default `false` - if `true`, this means
-a reboot is needed to apply the changes made by the role
+Default: `false`
 
-## Dependencies
+### `mssql_accept_microsoft_sql_server_2019_standard_eula`
 
-A list of other roles hosted on Galaxy should go here, plus any details in
-regards to parameters that may need to be set for other roles, or variables
-that are used from other roles.
+Set this variable to `true` to indicate that you accept EULA for using the
+`mssql-conf` utility.
 
-## Example Playbook
+The license terms for this product can be found in `/usr/share/doc/mssql-server`
+or downloaded from <https://go.microsoft.com/fwlink/?LinkId=2104078&clcid=0x409>.
+The privacy statement can be viewed at
+<https://go.microsoft.com/fwlink/?LinkId=853010&clcid=0x409>.
+
+Default: `false`
+
+### `mssql_password`
+
+The password for the database sa user. The password must have a minimum length
+of 8 characters, include uppercase and lowercase letters, base 10 digits or
+non-alphanumeric symbols. Do not use single quotes ('), double quotes ("), and
+spaces in the password because `sqlcmd` cannot authorise when the password
+includes those symbols.
```

Comment:
I'm just curious. Are the multi-byte characters accepted? In the microsoft world, UCS-2 is the standard encoding, I'd think.

And `aspell` checker says "authorise" should be "authorize", but I found "authorise" is used in UK!

Reply:
yes - authorize is en_US - authorise is en_GB

Reply:
re: multi-byte characters accepted

https://yaml.org/spec/1.2/spec.html#id2770814

So we may have to handle some sort of conversion to UCS-2

Reply:
hmm - thinking about this some more - we have the same problem with every other role - that is, how do we know that when the user inputs some Unicode, the underlying program being managed will accept it?  So I think we should not worry about this for now.